### PR TITLE
documentation #58: updated swagger for groups

### LINKED ISSUE #58: [DOC] Update Swagger for Groups [1]
We are updating the documentation for all endpoints and adding more _description and details_. 

One of the main controllers is Group. You should update the documentation according to the **pattern** below:
```ts
  @ApiBearerAuth()
  @ApiOkResponse()
  @ApiBadRequestResponse({
    description: `\n
    InvalidEntityIdException:
      User with such id is not found

    AlreadyRegisteredException:
      User is already registered

    InvalidBodyException:
      Group id can not be empty
      IsCaptain must be a boolean
      IsCaptain can not be empty
    
    AbsenceOfCaptainException:
      Captain was not found
    
    AbsenceOfCaptainTelegramException:
      Captain's telegramId was not found`,
  })
  @ApiUnauthorizedResponse({
    description: `\n
    UnauthorizedException:
      Unauthorized`,
  })
  @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action
    
    ForbiddenException:
      Forbidden`,
  })
  @ApiParam({
    name: 'userId',
    required: true,
    description: 'Id of a user to request to the group',
  })
  @ApiEndpoint({
    summary: 'Request captain or admin to join the group',
    permissions: PERMISSION.USERS_$USERID_GROUP_REQUEST,
  })
  @Patch('/:userId/requestNewGroup')
  requestNewGroup (
    @Param('userId', UserByIdPipe) userId: string,
    @Body() body: GroupRequestDTO,
  ) {
    return this.userService.requestNewGroup(userId, body);
  }
```
**Structure:**
- auth (if needed)
- ok response
- client errors
- params/query description
- api endpoint for summary, permissions, guards

You should also add **field description** in the body or response.
Add **summary** to endpoints.
Set **default values** for the optional fields

Image

_Buena suerte_

[Our Swagger](https://apidev.fictadvisor.com/api)
[Doc](https://docs.nestjs.com/openapi/introduction)

### PR DESCRIPTION
**Pull request by bohdanbulakh**
- updated addUnregistered endpoint
- updated verifyStudent endpoint
- updated moderatorSwitch endpoint

closes #58

### CODE REVIEW

File: fictadvisor-api/src/v2/api/controllers/GroupController.ts [hunk 1]
```diff
@@ -470,6 +489,21 @@ export class GroupController {
...
+    description: 'Id of the user to switch role',
+  })
+  @ApiEndpoint({
+    summary: 'Switch user role',
+    permissions: PERMISSION.GROUPS_$GROUPID_ADMIN_SWITCH,
+  })
+  @Patch('/:groupId/switch/:userId')
   async moderatorSwitch (
     @Param('groupId', GroupByIdPipe) groupId: string,
     @Param('userId', UserByIdPipe) userId: string,
```
Comment (SharpDevOps10) [inline fictadvisor-api/src/v2/api/controllers/GroupController.ts:509]: Add comma after `@Body() body: RoleDTO` => `@Body() body: RoleDTO,`

File: fictadvisor-api/src/v2/api/dtos/RoleDTO.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@ import { ApiProperty } from '@nestjs/swagger';
 export class RoleDTO {
   @ApiProperty({
     enum: RoleName,
+    description: 'User roles ',
```
Comment (SharpDevOps10) [inline fictadvisor-api/src/v2/api/dtos/RoleDTO.ts:9]: Redundant space : `'User roles'`

**Review by SharpDevOps10 — CHANGES_REQUESTED** [review]
Also remove redundant import `import { Access } from 'src/v2/security/Access';`

**Review by SharpDevOps10 — APPROVED** [review]
💋

**Review by fokaaas — APPROVED** [review]

**Pull request merged**